src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -733,6 +735,63 @@ export const deleteFormLogic = (
   })
 }
 
+/**
+ * Updates form logic.
+ * @param form The original form to update logic in
+ * @param logicId the logicId to update
+ * @param updatedLogic Object containing the updated logic
+ * @returns ok(updated logic dto) on success
+ * @returns err(database errors) if db error is thrown during logic update
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const updateFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+  updatedLogic: LogicDto,
+): ResultAsync<ILogicSchema, DatabaseError | LogicNotFoundError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic._id.toHexString() === logicId)) {
+    logger.error({
+      message: 'Error occurred - logicId to be updated does not exist',
+      meta: {
+        action: 'updateFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return errAsync(new LogicNotFoundError())
+  }
+
+  // Update specified logic
+  return ResultAsync.fromPromise(
+    FormModel.updateFormLogic(String(form._id), logicId, updatedLogic),
+    (error) => {
+      logger.error({
+        message: 'Error occurred when updating form logic',
+        meta: {
+          action: 'updateFormLogic',
+          formId: form._id,
+          logicId,
+          updatedLogic,
+        },
+        error,
+      })
+      return transformMongoError(error)
+    },
+    // On success, return updated form logic
+  ).andThen((updatedForm) => {
+    if (!updatedForm) {
+      return errAsync(new FormNotFoundError())
+    }
+    const updatedLogic = updatedForm.form_logics?.filter(
+      (logic) => logic._id.toHexString() === logicId,
+    )[0]
+    if (!updatedLogic) return errAsync(new LogicNotFoundError()) // Possible race condition if logic gets deleted after the initial logicId check but before the db update
```

Comment:
possible to implement a `getLogicById`, similar to `getFormFieldById`?

Reply:
thanks, edited